packages/contractkit/src/web3-contract-cache.ts

```diff
@@ -147,21 +147,24 @@ export class Web3ContractCache {
   /**
    * Get native web3 contract wrapper
    */
-  async getContract<C extends keyof typeof ContractFactories>(contract: C, address?: string) {
+  async getContract<
+    TContractType extends keyof typeof ContractFactories,
+    TContract = ReturnType<typeof ContractFactories[TContractType]>
+  >(contract: TContractType, address?: string): Promise<TContract> {
     if (this.cacheMap[contract] == null || address !== undefined) {
       debug('Initiating contract %s', contract)
       const createFn = ProxyContracts.includes(contract)
```

Comment:
I don't understand this ternary if here -- it looks a bit weird, irrespective of me removing that unneeded typecast:

```
      const createFn = ProxyContracts.includes(contract)
        ? newProxy
        : ContractFactories[contract]
        ? ContractFactories[contract]
        : newProxy
```

So if ProxyContracts.includes(contract) is true, then use "newProxy", else if ContractFactories[contract] exists then use that otherwise use newProxy?